crates/sol-macro/src/expand/contract.rs

```diff
@@ -429,14 +425,5 @@ fn generate_variant_methods((variant, ty): (&Ident, &Ident)) -> TokenStream {
                 _ => ::core::option::Option::None,
             }
         }
-
-        #[doc = #try_into_variant_doc]
-        #[inline]
-        pub const fn #try_into_variant(self) -> ::core::result::Result<#ty, Self> {
-            match self {
-                Self::#variant(inner) => ::core::result::Result::Ok(inner),
-                _ => ::core::result::Result::Err(self),
-            }
-        }
```

Comment:
cannot be const if destructor cannot be evaluated, so generating this function is useless when we are already generating `TryFrom`